Makefile

```diff
@@ -0,0 +1,19 @@
+PRE_COMMIT_VERSION ?= 4.3.0
+
+VENV := .venv
+PIP := $(VENV)/bin/python -m pip
+PRE_COMMIT := $(VENV)/bin/pre-commit
+
+.PHONY: init lint check clean update
+
+init:
+	@test -d $(VENV) || python3 -m venv $(VENV)
+	$(PIP) install "pre-commit==$(PRE_COMMIT_VERSION)"
```

Comment:
You don't have to abstract pip. Since we're using a venv it's unlikely to change and by making an abstraction you're simply making the line less readable. If it was used multiple places in the file it would be a different case but for now I would keep things as simple as possible and therefor not use a variable